include/seqan3/range/view/translate_join.hpp

```diff
@@ -0,0 +1,405 @@
+    {
+        assert(n < size());
+        unsigned index_frame = n % selected_frames.size();
+        unsigned index_urange = (n - index_frame) / selected_frames.size();
+        return urange[index_urange] | view::translate_single(selected_frames[index_frame]);
+    }
+
+    //!\overload
+    const_reference operator[](size_type const n) const
+    {
+        assert(n < size());
+        unsigned index_frame = n % selected_frames.size();
+        unsigned index_urange = (n - index_frame) / selected_frames.size();
+        return urange[index_urange] | view::translate_single(selected_frames[index_frame]);
+    }
+    //!\}
+};
+
+//!\brief Class template argument deduction for view_translate_join.
+template <typename urng_t>
+//!\cond
+    requires std::ranges::SizedRange<urng_t> &&
+             std::ranges::RandomAccessRange<urng_t> &&
+             NucleotideAlphabet<innermost_value_type_t<urng_t>>
+//!\endcond
+view_translate_join(urng_t &&, translation_frames const = translation_frames{}) -> view_translate_join<std::ranges::all_view<urng_t>>;
+
+} // namespace seqan3::detail
+
+// ============================================================================
+//  translate (adaptor object)
+// ============================================================================
+
+namespace seqan3::view
+{
+
+/*!\name Alphabet related views
+ * \{
+ */
+
+/*!\brief A view that translates nucleotide into aminoacid alphabet with 1, 2, 3 or 6 frames. Input and output range are always two-dimensional.
+ * \tparam urng_t The type of the range being processed.
+ * \param[in] urange The range being processed. Needs to be a range of ranges (two-dimensional).
+ * \param[in] tf A value of seqan3::tanslation_frames that indicates the desired frames.
+ * \returns A range of ranges containing frames with aminoacid sequence. See below for the properties of the returned range.
+ * \ingroup view
+ *
+ * \details
+ *
+ * This view can be used to translate nucleotide sequences into aminoacid sequences (see translation_frames for possible combination of frames).
+ * This view only operates on two-dimensional input (range of ranges) and outputs a range of ranges no matter the number of input sequences or
+ * the number of translation frames given. Therefore, it has the same capabilities as the standard view_translate but concatenates the different
+ * frames of the different input sequences rather than having a separate range for each input sequence containing the translated frames. In the output,
+ * frames are ordered in a way, that all requested frames are listed per sequence directly after each other in the order of input sequences.
+ * improved and efficient downstream post-processing if needed. However, the index of a frame for a specific sequence needs to be calculated via
+ * modulo operations in this case. The i-th frame of the j-th sequence can be calculated by n = (i * s) + j, where s is the number of frames used
+ * for translation (index starting at zero).
```

Comment:
I would add a paragraph:

In short, this views behaves the same as:
```cpp
std::vector<std::vector<dna5>> vec = /* ... */;
auto v = vec | view::translate | std::view::join;
```
Except that the performance is better and the returned range still models std::ranges::RandomAccessRange and std::ranges::SizedRange.